### PR TITLE
feat(action-group, action): add grid layout to action-group. add center layout to action

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #1506

## Summary
Adds a grid layout option to ActionGroup.
Also adds a related center layout option to Action.
Note: centered layout in Action only applies when text-enabled is false.

Enables layouts like this:
image

### CODE REVIEW

**Comment by asangma** [comment]
I know we wanted to stop using attribute selectors for our styling and instead apply classes based on those attributes.
I will need to add a node to receive those classes and would prefer to do that in another pass so as to reduce the risk of regressions at this time.

File: src/components/calcite-action-group/calcite-action-group.tsx [hunk 1]
```diff
@@ -24,6 +25,16 @@ export class CalciteActionGroup {
...
 
+  /**
+   * Indicates the horizontal, vertical, or grid layout of the component.
+   */
+  @Prop({ reflect: true}) layout?: Layout = "vertical";
+
+  /**
+   * Indicates number of columns.
+   */
+  @Prop({ reflect: true}) columnCount?: ColumnCount;
```
Comment (driskull) [inline src/components/calcite-action-group/calcite-action-group.tsx:36]: nitpick: maybe just `columns`?

File: src/components/calcite-action-group/calcite-action-group.tsx [hunk 2]
```diff
@@ -24,6 +25,16 @@ export class CalciteActionGroup {
    */
   @Prop({ reflect: true }) expanded = false;
 
+  /**
+   * Indicates the horizontal, vertical, or grid layout of the component.
+   */
+  @Prop({ reflect: true}) layout?: Layout = "vertical";
```
Comment (driskull) [inline src/components/calcite-action-group/calcite-action-group.tsx:31]: nitpick: The property is optional but has a default value so the optional isn't needed.

**Review by driskull — APPROVED** [review]
Looks good. Can we get some tests first? or a follow up issue for tests?

File: src/components/calcite-action-group/calcite-action-group.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-action-group/calcite-action-group.tsx:36]: I like it. Thanks!

File: src/components/calcite-action-group/calcite-action-group.tsx (hunk 2 above)
Comment (asangma) [reply src/components/calcite-action-group/calcite-action-group.tsx:31]: Ah...good catch.

**Pull request merged**